CMakeLists.txt

```diff
@@ -1,39 +1,57 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.16)
 
-project(mpifx VERSION 0.1 LANGUAGES Fortran)
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+include(${CMAKE_CURRENT_SOURCE_DIR}/config.cmake)
 
-set(LIBRARY_ONLY FALSE CACHE BOOL "Whether only library should be compiled")
+project(MpiFx VERSION 0.1 LANGUAGES Fortran)
 
-option(BUILD_SHARED_LIBS "Whether the library should be a shared one" FALSE)
-
-option(INSTALL_INCLUDE_FILES "Whether include / module files should be installed" TRUE)
-
-# Installation paths
-set(INSTALL_BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH
-  "Installation directory for executables")
-
-set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH
-  "Installation directory for libraries")
-
-set(INSTALL_MOD_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH
-  "Installation directory for Fortran module files")
-
-set(INSTALL_CMAKE_DIR "${CMAKE_INSTALL_PREFIX}/lib/cmake" CACHE PATH
-  "Installation directory for CMake package export files")
-
-option(BUILD_SHARED_LIBS "Whether the library should be shared" FALSE)
-
-option(INSTALL_INCLUDE_FILES "Whether include and module files should be installed" TRUE)
+include(MpiFxUtils)
 
+#
+# Prerequisites
+#
 find_package(MPI REQUIRED)
 find_program(FYPP fypp)
-if(FYPP)
-  message(STATUS "Preprocessor fypp: ${FYPP}")
-else()
-  message(FATAL_ERROR "Prepropcessor fypp not found")
+if(NOT FYPP)
+  message(FATAL_ERROR "Preprocessor fypp could not be found")
 endif()
 
+#
+# Build instructions
+#
 add_subdirectory(lib)
-if(NOT LIBRARY_ONLY)
+if(NOT BUILD_LIBRARY_ONLY)
   add_subdirectory(test)
 endif()
+
+#
+# Install
+#
+install(EXPORT mpifx-targets
+  FILE mpifx-targets.cmake
+  NAMESPACE MpiFx::
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/mpifx")
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+  ${CMAKE_SOURCE_DIR}/utils/cmake/mpifx-config.cmake.in
+  ${CMAKE_BINARY_DIR}/cmake/mpifx-config.cmake
```

Comment:
Any reason for not using `CMAKE_CURRENT_SOURCE_DIR` and `CMAKE_CURRENT_BINARY_DIR` here? `CMAKE_SOURCE_DIR` would point to the source directory of the root project, which is not necessarily `mpifx` when used in a subproject, or am I mistaken?

Reply:
Absolutely correct, thx!